aws/resource_aws_appstream_fleet.go

```diff
@@ -0,0 +1,588 @@
+		input.IdleDisconnectTimeoutInSeconds = aws.Int64(int64(d.Get("idle_disconnect_timeout_in_seconds").(int)))
+	}
+
+	if d.HasChange("display_name") {
+		input.DisplayName = aws.String(d.Get("display_name").(string))
+	}
+
+	if d.HasChange("image_name") {
+		input.ImageName = aws.String(d.Get("image_name").(string))
+	}
+
+	if d.HasChange("image_arn") {
+		input.ImageArn = aws.String(d.Get("image_arn").(string))
+	}
+
+	if d.HasChange("iam_role_arn") {
+		input.IamRoleArn = aws.String(d.Get("iam_role_arn").(string))
+	}
+
+	if d.HasChange("stream_view") {
+		input.StreamView = aws.String(d.Get("stream_view").(string))
+	}
+
+	if d.HasChange("instance_type") {
+		input.InstanceType = aws.String(d.Get("instance_type").(string))
+	}
+
+	if d.HasChange("max_user_duration_in_seconds") {
+		input.MaxUserDurationInSeconds = aws.Int64(int64(d.Get("max_user_duration_in_seconds").(int)))
+	}
+
+	if d.HasChange("vpc_config") {
+		input.VpcConfig = expandVpcConfig(d.Get("vpc_config").([]interface{}))
+	}
+
+	resp, err := conn.UpdateFleetWithContext(ctx, input)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error updating Appstream Fleet (%s): %w", d.Id(), err))
+	}
+
+	if d.HasChange("tags") {
+		arn := aws.StringValue(resp.Fleet.Arn)
+
+		o, n := d.GetChange("tags")
+		if err := keyvaluetags.AppstreamUpdateTags(conn, arn, o, n); err != nil {
+			return diag.FromErr(fmt.Errorf("error updating Appstream Fleet tags (%s): %w", d.Id(), err))
+		}
+	}
+
+	// Start fleet workflow
+	_, err = conn.StartFleetWithContext(ctx, &appstream.StartFleetInput{
+		Name: aws.String(d.Id()),
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error starting Appstream Fleet (%s): %w", d.Id(), err))
+	}
+
+	if _, err = waiter.FleetStateRunning(ctx, conn, d.Id()); err != nil {
+		return diag.FromErr(fmt.Errorf("error waiting for Appstream Fleet (%s) to be running: %w", d.Id(), err))
+	}
+
+	return resourceAwsAppStreamFleetRead(ctx, d, meta)
+
+}
+
+func resourceAwsAppStreamFleetDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).appstreamconn
+
+	// Stop fleet workflow
+	_, err := conn.StopFleetWithContext(ctx, &appstream.StopFleetInput{
+		Name: aws.String(d.Id()),
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error stopping Appstream Fleet (%s): %w", d.Id(), err))
+	}
+
+	if _, err = waiter.FleetStateStopped(ctx, conn, d.Id()); err != nil {
+		return diag.FromErr(fmt.Errorf("error waiting for Appstream Fleet (%s) to be stopped: %w", d.Id(), err))
+	}
+
+	_, err = conn.DeleteFleetWithContext(ctx, &appstream.DeleteFleetInput{
+		Name: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		if tfawserr.ErrCodeEquals(err, appstream.ErrCodeResourceNotFoundException) {
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error deleting Appstream Fleet (%s): %w", d.Id(), err))
+	}
+	return nil
+
+}
+
+func expandComputeCapacity(computeCapacity []interface{}) *appstream.ComputeCapacity {
```

Comment:
We prefer that all the expanders and flatteners use generic parameter and variable names such as in the examples here: https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/data-handling-and-conversion.md#expand-functions-for-blocks

In other words, `computeCapacity` becomes `tfList`, `computeConf` becomes `apiObject`, etc.